### PR TITLE
CMSiteConfiguration- Adds a resource for CMSite

### PR DESCRIPTION
**Pull request by NEllis280**
#### Pull Request (PR) description
This PR adds a resource for CMSite.

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/103)

### CODE REVIEW

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 [hunk 1]
```diff
@@ -0,0 +1,1232 @@
...
+        $useCrl = $true
+        $pkiClient = $true
+        $sccmCert = $true
+    }
+
+    foreach ($item in $comSec.Props)
+    {
+        switch ($item.PropertyName)
+        {
+            'Certificate Store'              { if ([string]::IsNullOrEmpty($item.Value1))
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:138]: The other instance of switch with if has the if on a new line, either move that to top line of move this to new line to match

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 [hunk 2]
```diff
@@ -0,0 +1,1232 @@
...
+            if (($collectionMax -ne 0) -and ($collectionMax -le $collectionDefault))
+            {
+                throw ($script:localizedData.CollectionError -f $collectionDefault, $collectionMax)
+            }
+        }
+
+        if ($state.SiteType -eq 'Primary')
+        {
+            $defaultValues += @('ClientCheckCertificateRevocationListForSiteSystem','UsePkiClientCertificate',
+            'RequireSigning','UseEncryption','EnableLowFreeSpaceAlert','EnableWakeOnLan','ClientCertificateCustomStoreName',
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:621]: line 587 has the additional stuff under the array indented. line 621 and all lines till the close ) should be indented from the array specifications for uniform look and know that its still part of the array. here and the one below as well

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 [hunk 3]
```diff
@@ -0,0 +1,1232 @@
...
+                    if ($PSBoundParameters.ContainsKey('FreeSpaceThresholdWarningGB') -or
+                        $PSBoundParameters.ContainsKey('FreeSpaceThresholdCriticalGB'))
+                    {
+                        Write-Warning -Message $script:localizedData.IgnoreAlertsSettings
+                    }
+                }
+            }
+
+            $wolParams = @('WakeOnLanTransmissionMethodType','RetryNumberOfSendingWakeupPacketTransmission','SendingWakeupPacketTransmissionDelayMins',
+            'MaximumNumberOfSendingWakeupPacketBeforePausing','SendingWakeupPacketBeforePausingWaitSec','ThreadNumberOfSendingWakeupPacket',
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:668]: same comment as above

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+                if ($PSBoundParameters.ContainsKey('FreeSpaceThresholdWarningGB') -or
+                    $PSBoundParameters.ContainsKey('FreeSpaceThresholdCriticalGB'))
+                {
+                    Write-Warning -Message $script:localizedData.IgnoreAlertsSettings
+                }
+            }
+        }
+
+        $wolParams = @('WakeOnLanTransmissionMethodType','RetryNumberOfSendingWakeupPacketTransmission','SendingWakeupPacketTransmissionDelayMins',
+        'MaximumNumberOfSendingWakeupPacketBeforePausing','SendingWakeupPacketBeforePausingWaitSec','ThreadNumberOfSendingWakeupPacket',
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:1149]: indent like above for the line 1100

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,1594 @@
...
+                        @{
+                            PropertyName = 'MaxThreads'
+                            Value        = 3
+                        }
+                        @{
+                            PropertyName = 'SendMode'
+                            Value        = 1
+                        }
+                    )
+
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:454]: extra line

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,1594 @@
...
+                        @{
+                            PropertyName = 'MaxThreads'
+                            Value        = 3
+                        }
+                        @{
+                            PropertyName = 'SendMode'
+                            Value        = 2
+                        }
+                    )
+
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:485]: extra line

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,1594 @@
...
+                    $result.RetryNumberOfSendingWakeupPacketTransmission      | Should -Be -ExpectedValue $null
+                    $result.SendingWakeupPacketTransmissionDelayMins          | Should -Be -ExpectedValue $null
+                    $result.MaximumNumberOfSendingWakeupPacketBeforePausing   | Should -Be -ExpectedValue $null
+                    $result.SendingWakeupPacketBeforePausingWaitSec           | Should -Be -ExpectedValue $null
+                    $result.ThreadNumberOfSendingWakeupPacket                 | Should -Be -ExpectedValue $null
+                    $result.SendingWakeupPacketTransmissionOffsetMins         | Should -Be -ExpectedValue 0
+                    $result.ClientCertificateCustomStoreName                  | Should -Be -ExpectedValue 'SMSStore'
+                    $result.TakeActionForMultipleCertificateMatchCriteria     | Should -Be -ExpectedValue 'FailSelectionAndSendErrorMessage'
+                    $result.ClientCertificateSelectionCriteriaType            | Should -Be -ExpectedValue 'ClientAuthentication'
+                    $result.ClientCertificateSelectionCriteriaValue           | Should -Be -ExpectedValue ''
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:574]: in get doesn't this default to Personal?

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,1594 @@
...
+                    $result.SendingWakeupPacketTransmissionOffsetMins         | Should -Be -ExpectedValue 10
+                    $result.ClientCertificateCustomStoreName                  | Should -Be -ExpectedValue 'Personal'
+                    $result.TakeActionForMultipleCertificateMatchCriteria     | Should -Be -ExpectedValue 'FailSelectionAndSendErrorMessage'
+                    $result.ClientCertificateSelectionCriteriaType            | Should -Be -ExpectedValue 'CertificateSubjectOrSanIncludesAtrributes'
+                    $result.ClientCertificateSelectionCriteriaValue           | Should -Be -ExpectedValue 'Test'
+                }
+            }
+        }
+
+        Describe 'ConfigMgrCBDsc - DSC_CMSiteConfiguration\Set-TargetResource' -Tag 'Set'{
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:923]: need space between -Tag 'Set' and the {

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,1594 @@
...
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+#Begin Testing
+try
+{
+    InModuleScope $script:dscResourceName {
+        Describe 'ConfigMgrCBDsc - DSC_CMSiteConfiguration\Get-TargetResource' -Tag 'Get'{
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:39]: need space between 'Get' and {

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,1594 @@
...
+                        SiteCode                     = 'Lab'
+                        FreeSpaceThresholdWarningGB  = 20
+                        FreeSpaceThresholdCriticalGB = 10
+                    }
+
+                    $ignoreSMSCert = @{
+                        SiteCode                        = 'Lab'
+                        UseSmsGeneratedCert             = $true
+                        ClientComputerCommunicationType = 'HttpsOnly'
+
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:1035]: extra line

File: tests/Unit/CMSiteConfiguration.tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,1594 @@
...
+                    { Set-TargetResource @inputMissingType } | Should -Throw -ExpectedMessage $missingCertType
+                    Assert-MockCalled Import-ConfigMgrPowerShellModule -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-CMSite -Exactly -Times 0 -Scope It
+                }
+            }
+        }
+
+        Describe 'ConfigMgrCBDsc - DSC_CMSiteConfiguration\Test-TargetResource' -Tag 'Test'{
```
Comment (jeffotterpohl) [inline tests/Unit/CMSiteConfiguration.tests.ps1:1285]: need a space between 'Test' and {

File: tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1 [hunk 12]
```diff
@@ -3111,6 +3111,206 @@ InModuleScope $script:subModuleName {
...
+                        }
+                        @{
+                            Name         = 'UseEncryption'
+                            PropertyType = '[bool]'
+                            IsMandatory  = $false
+                            Values       = '{}'
+                        }
+                        @{
+                            Name         = 'MaximumConcurrentSendingForAllSite'
+                            PropertyType = '[UInt32]'
```
Comment (jeffotterpohl) [inline tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1:3176]: i know this is throughout, but this should be lower case like string and bool should be uint32 in this section

File: tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1 [hunk 13]
```diff
@@ -5912,6 +6112,66 @@ InModuleScope $script:subModuleName {
...
+                RetryNumberOfSendingWakeupPacketTransmission      = 1
+                SendingWakeupPacketTransmissionDelayMins          = 10000
+                MaximumNumberOfSendingWakeupPacketBeforePausing   = 10
+                SendingWakeupPacketBeforePausingWaitSec           = 3
+                ThreadNumberOfSendingWakeupPacket                 = 10
+                SendingWakeupPacketTransmissionOffsetMins         = 10
+                ClientCertificateCustomStoreName                  = 'SMSStore'
+                TakeActionForMultipleCertificateMatchCriteria     = 'SelectCertificateWithLongestValidityPeriod'
+                ClientCertificateSelectionCriteriaType            = 'ClientAuthentication'
+                ClientCertificateSelectionCriteriaValue           = 'Personal'
```
Comment (jeffotterpohl) [inline tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1:6153]: missing Configurationname, DependsOn, ModuleName, and several other settings if this is being returned by Invoke-DSCResource not required but would be uniform settings and would be returned by the invoke.  They are in the $invokeSiteConfigurationCas

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 (hunk 1 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:138]: Fixed in local repo.

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 (hunk 2 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:621]: Fixed in local repo.

File: source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1 (hunk 4 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMSiteConfiguration/DSC_CMSiteConfiguration.psm1:1149]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 5 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:454]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 6 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:485]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 8 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:923]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 10 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:1035]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 11 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:1285]: Fixed in local repo.

File: tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1 (hunk 13 above)
Comment (NEllis280) [reply tests/Unit/ConfigMgrCBDsc.ReverseDsc.tests.ps1:6153]: Fixed in local repo.

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 9 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:39]: resolved in local repo

File: tests/Unit/CMSiteConfiguration.tests.ps1 (hunk 7 above)
Comment (NEllis280) [reply tests/Unit/CMSiteConfiguration.tests.ps1:574]: That's the ClientCertificateCustomStoreName parameter you're thinking of. It does default to personal. The test for that is captured on line 913.

**Review by jeffotterpohl — APPROVED** [review]

**Pull request merged**